src/multihash.rs

```diff
@@ -54,12 +54,41 @@ impl<const S: usize> Default for Multihash<S> {
     }
 }
 
+/// Returned from [`Multihash::new`]
+#[derive(Debug)]
+#[non_exhaustive]
+pub enum CreationError {
+    /// The digest exceeds the capacity of the multihash
+    DigestTooBig,
+}
```

Comment:
I think I'd prefer a struct instead of an enum with a single variant, plus we should print the same message as `Error::invalid_size`.